script.js

```diff
@@ -0,0 +1,181 @@
+    ],
+  },
+  {
+    question: "Which JavaScript statement is incorrect?",
+    options: [
+      { text: '"b" + "a" + + "a" + "a" //output "baNaNa"', isCorrect: false },
+      { text: "Number.MIN_VALUE > 0 //true", isCorrect: true },
+      { text: "functions can execute themselves", isCorrect: false },
+      { text: "undefined never can be defined", isCorrect: false },
+    ],
+  },
+  {
+    question: "Which JavaScript syntax is correct?",
+    options: [
+      { text: 'return {""};', isCorrect: false },
+      { text: '1 + 2 + "3" //output "123"', isCorrect: false },
+      { text: "[,*,].length", isCorrect: false },
+      { text: ";let a = 2", isCorrect: true },
+    ],
+  },
+  {
+    question: "How many years in the row JavaScript keep being the most popular programming language between developers?",
+    options: [
+      { text: "10 years", isCorrect: false },
+      { text: "11 years", isCorrect: true },
+      { text: "12 years", isCorrect: false },
+      { text: "13 years", isCorrect: false },
+    ],
+  },
+];
+
+function filterQuestions() {
+  const contentInput = document.getElementById("content-input").value;
+  const filteredQuestions = quizData.filter((question) =>
+    question.question.toLowerCase().includes(contentInput.toLowerCase())
+  );
+
+  const filteredQuestionsDiv = document.querySelector(".filteredQuestions");
+  filteredQuestionsDiv.innerHTML = "";
+
+  if (filteredQuestions.length === 0) {
+    const noResultMessage = document.createElement("p");
+    noResultMessage.textContent = "No questions found";
+    filteredQuestionsDiv.appendChild(noResultMessage);
+  } else {
+    filteredQuestions.forEach((question) => {
+      const questionElement = document.createElement("div");
+      questionElement.classList.add("question");
+      
+      const questionText = document.createElement("p");
+      questionText.textContent = question.question;
+      questionText.style.fontWeight = "bold";
+      questionElement.appendChild(questionText);
+      questionElement.style.marginTop = "10px";
+
+      const optionsElement = document.createElement("ol");
+      optionsElement.classList.add("options");
+      optionsElement.style.padding = "10px";
+
+      question.options.forEach((option) => {
+        const optionElement = document.createElement("li");
+        optionElement.textContent = option.text;
+        optionsElement.appendChild(optionElement);
+      });
+
+      questionElement.appendChild(optionsElement);
+      filteredQuestionsDiv.appendChild(questionElement);
+    });
+  }
+}
+
+function startQuiz() {
+  const player1 = document.getElementById('name-input1').value;
+  const player2 = document.getElementById('name-input2').value;
+  
+  const playersList = document.getElementById('playersList');
+  playersList.innerHTML = 
+  '<div>' + player1 + '<span id="scorePlayer1">0</span><button onclick="addScore(1, true)">Correct</button><button onclick="addScore(1, false)">Wrong</button></div>' +
```

Comment:
When clicking these buttons, The page scrolls to the required question input. You can get around this by giving the buttons a type, like this:

`<button type="button" onclick="doButtonStuff()">Click Me</button>`